### PR TITLE
Fix afls command family

### PR DESCRIPTION
**Pull request by salutdamour**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

issue #238 

| afls same as aflsa
| aflsa sort by address (same as afls)
| aflss sort by size
| aflsn sort by name
| aflsb sort by number of basic blocks

**Test plan**

Used the `bins/elf/ls` to test.

**Closing issues**

...

### CODE REVIEW

**Comment by salutdamour** [comment]
Sorry. Still failed on clang-format. I have changed cmd_analysis.c on Ubuntu18, so I use the command `clang-format -style=file -i ~/rizin/librz/core/cmd_analysis.c` to format my code. However, it takes more problems compared with the last time.

**Comment by wargio** [comment]
just run `./sys/clang-format.py` from the root folder of rizin

**Comment by salutdamour** [comment]
> just run `./sys/clang-format.py` from the root folder of rizin

I only modified a C file.When I ran the clang-format.py, it changed 34 files. Is it normal?  I am afraid that I can’t run the clang-format test of PR.

**Comment by XVilka** [comment]
@ReJimp it might be because you have too old clang version.

**Comment by salutdamour** [comment]
@XVilka I think the file `clang-format.py` needs to be updated. I installed `clang-format-11` on Ubuntu20. When I run the script, it pointed out that I didn't install clang-format. When I used `sudo apt search clang-format
`, it showed that `clang-format` 's highest version is 10. `clang-format` on the 76th line and 78th line in  the `clang-format.py` replaced by `clang-format-11` is better.

**Comment by wargio** [comment]
i suggest to add a symlink to `clang-format` since ubuntu calls it `clang-format-11`.
that said my clang-format version `clang-format version 11.1.0` does not produce that result.

**Comment by wargio** [comment]
be sure to commit and push the changes.

File: librz/core/cmd_analysis.c [hunk 1]
```diff
@@ -757,16 +757,16 @@ static int cmpname(const void *_a, const void *_b) {
 static int cmpsize(const void *a, const void *b) {
 	ut64 sa = (int)rz_analysis_function_linear_size((RzAnalysisFunction *)a);
 	ut64 sb = (int)rz_analysis_function_linear_size((RzAnalysisFunction *)b);
-	return (sa > sb) ? -1 : (sa < sb) ? 1
-					  : 0;
+	return (sa > sb) ? 1 : (sa < sb) ? 1
```
Comment (XVilka) [inline librz/core/cmd_analysis.c]: I am not sure this change is right. Why it does return `1` in both cases? This is not how comparison callback works...

**Review by XVilka — CHANGES_REQUESTED** [review]

**Comment by XVilka** [comment]
Could you please add tests for this?

**Comment by salutdamour** [comment]
Does it mean that show the test results or write a test in test/？What's more, I am also interested in the issue #613. Can I try solving it at the same time? If without your help, I may waste more time on the envrionment. Just when I felt I was on track, it was time for the RSOC review.

**Comment by XVilka** [comment]
@ReJimp Yes, please add a few tests in `test/db/cmd/cmd_afl`.

**Comment by XVilka** [comment]
@ReJimp sure, you can do that, take https://github.com/rizinorg/rizin/issues/613

File: librz/core/cmd_analysis.c [hunk 2]
```diff
@@ -2908,22 +2908,27 @@ RZ_IPI int rz_cmd_analysis_fcn(void *data, const char *input) {
 			case 'a': // "aflsa"
 				core->analysis->fcns->sorted = false;
 				rz_list_sort(core->analysis->fcns, cmpaddr);
+				rz_core_analysis_fcn_list(core, NULL, "o");
```
Comment (ret2libc) [inline librz/core/cmd_analysis.c:2911]: why `"o"`?

Comment (salutdamour) [reply librz/core/cmd_analysis.c:2911]: I think it can jump to the default branch in `rz_core_analysis_fcn_list` as same as the command `afl`.

Comment (ret2libc) [reply librz/core/cmd_analysis.c:2911]: Ok, good enough for now! That `o` does not have any particular meaning, but considering it's already used elsewhere it is fine for now, thanks!

**Review by ret2libc — COMMENTED** [review]
@XVilka @thestr4ng3r @Maijin `afls` commands right now change the ordering of the function list, so if you do `aflsa`, then `afl` will show the new ordering. With this PR we are slightly changing the behaviour of these `afls` commands to also print the functions list immediately. Do we want to keep this side-effect behaviour?

What I mean is that if you do:
```
> afl
> aflsn
> afl
```
the two `afl` commands will show commands in a different order, just because `aflsn` was called in-between. WDYT? Maybe `aflsn` should not affect the ordering of the global function list.

**Comment by XVilka** [comment]
@ret2libc yes, agree - definitely it **should not** change the global ordering.

**Comment by salutdamour** [comment]
I think so. I will follow up on this issue.

**Comment by XVilka** [comment]
I am closing this since there was no progress and updates for months. Feel free to reopen if you update the PR.

**Pull request closed without merging**